src/firecracker/tests/verify_dependencies.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+use std::path::Path;
+
+use cargo_toml::{Dependency, DependencyDetail, DepsSet, Manifest};
+use regex::Regex;
+
+#[test]
+fn test_no_comparison_requirements() {
+    // HashMap mapping crate -> [(violating dependency, specified version)]
+    let mut violating_dependencies = HashMap::new();
+
+    let src_firecracker_path = match std::env::var("CARGO_MANIFEST_DIR") {
+        Ok(path) => path,
+        Err(_) => return, /* when running under kcov, this variable is not set. As we do not
+                           * actually run any firecracker code, we can just skip this test
+                           * without affecting coverage */
+    };
+    let src_path = format!("{}/..", src_firecracker_path);
+
+    for fc_crate in std::fs::read_dir(src_path).unwrap() {
+        let fc_crate = fc_crate.unwrap();
+        if fc_crate.metadata().unwrap().is_dir() {
+            let violating_in_crate =
+                violating_dependencies_of_cargo_toml(fc_crate.path().join("Cargo.toml"));
+
+            if !violating_in_crate.is_empty() {
+                violating_dependencies.insert(
+                    fc_crate.file_name().into_string().unwrap(),
+                    violating_in_crate,
+                );
+            }
+        }
+    }
+
+    assert_eq!(
+        violating_dependencies,
```

Comment:
Is this meant to check that `violating_dependencies` is empty? Can t we say `assert_eq(violating_dependencies.keys.len(), 0)`? -> just seems more straightforward

Reply:
Yes, this just checks that the list is empty. The reason I chose to write the test this way instead of `assert!(violating_deps.is_empty())` is that this way, if the list is not empty, the assert message will contain every violating dependency, whereas otherwise it will just tell you _that_ the assert failed, without telling you what dependencies caused it to fail.

Reply:
Nice, did not think of that!